### DISCUSSION TITLE
Ability to pass args to on_click or on_tap

### DISCUSSION THREAD

**Discussion by kspviswa** [flet-dev/flet#3589] [Q&A] [answered]
### Question

I'm try to pass args to event handlers for either ft.IconButton's on_click or ft.DataCell's on_tap handler. Is it possible ?

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by kspviswa** [comment, marked as answer]
I think I found a way. We can add value to `key` and then inspect it as part of `ft.ControlEvent's key` on the callback

**Comment by ndonkoHenri** [comment]
You can use lambda functions:
```py
on_click=lambda e: my_func(e, arg1, arg2, argx)
```

Obviously, if your function doesnt need `e`, you can remove it from the function call.
